pkg/network/ebpf/c/tracer-stats.h

```diff
@@ -51,6 +51,28 @@ static __always_inline void update_conn_stats(conn_tuple_t *t, size_t sent_bytes
         return;
     }
 
+    protocol_t local_protocol = PROTOCOL_UNCLASSIFIED;
+    conn_tuple_t t2 = *t;
+    // The classifier is a socket filter and there we are not accessible for pid and netns.
+    // The key is based of the source & dest addresses and ports, and the metadata.
+    t2.netns = 0;
+    t2.pid = 0;
+
+    protocol_t *protocol = bpf_map_lookup_elem(&connection_protocol, &t2);
```

Comment:
Assuming egress flow, the protocol is indexed by a `conn_tuple_t` object created in the socket filter post-NAT. At this point you are using a `conn_tuple_t` created pre-NAT. These two maybe different for example when dealing with a containerized application. Do you think this could cause accuracy issues since we will not be able to tag the stats object with the correct protocol?

Reply:
ok, so I ran the following:

- Run a docker with server in it (`docker run -it --rm -p 8082:8080 python:3.9-alpine python3 -m http.server 8080`)
- Run a tracer test with a client sending a http request to the server `http://127.0.0.1:8082`

I see that we capture 4 connections:

1. `[172.17.0.1:57474 ⇄ 172.17.0.5:8080]`
2. `[172.17.0.5:8080 ⇄ 172.17.0.1:57474]`
3. `[127.0.0.1:8082 ⇄ 127.0.0.1:37086]`
4. `[127.0.0.1:37086 ⇄ 127.0.0.1:8082]`

 (my dockers have the ip address of 172.17.0.0/16)

(that's without the change we've talked about)

in that scenario, I think the current implementation is sufficient

WDYT?

Reply:
Yes I agree. Looks correct now.

Reply:
I don't believe there is any NAT going on here; docker has a local proxy when you expose container ports, which is what is showing up here.

Reply:
Ok, run a test with explicit NAT and I've found the problem you are talking about
added a workaround to support NAT (using `tracepoint/net/net_dev_queue`) to allow us to correlate between `__sk_buff` in the `socket filter` to the `struct sock*` that is being used in the `kprobes` of `npm`